src/lib.rs

```diff
@@ -1047,9 +1047,33 @@ pub struct WorkspaceEditClientCapabilities {
     ///
     /// @since 3.16.0
     #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(deserialize_with = "deserialize_change_annotation_support")]
     pub change_annotation_support: Option<ChangeAnnotationWorkspaceEditClientCapabilities>,
 }
 
+fn deserialize_change_annotation_support<'de, D>(
+    d: D,
+) -> Result<Option<ChangeAnnotationWorkspaceEditClientCapabilities>, D::Error>
+where
+    D: de::Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum Repr {
+        Bool(bool),
+        Obj(ChangeAnnotationWorkspaceEditClientCapabilities),
+    }
+    Option::<Repr>::deserialize(d).map(|repr| {
+        repr.and_then(|repr| match repr {
+            Repr::Bool(true) => Some(ChangeAnnotationWorkspaceEditClientCapabilities {
+                groups_on_labels: None,
```

Comment:
This needs to be `groups_on_label`. It looks like I messed this up.

Reply:
Probably, though the err message would be different from the one i https://github.com/rust-analyzer/rust-analyzer/issues/6979 ?

Reply:
I'm wondering if that happens without this change: the client sends down the field with the wrong name (so we're not hitting the `Repr::Bool` case) and it fails to deserialize and bubbles up to a 'missing' error.

Reply:
Ah, that might be the case